src/Storages/StorageDistributed.cpp

```diff
@@ -1025,6 +1139,7 @@ QueryTreeNodePtr buildQueryTreeDistributed(SelectQueryInfo & query_info,
         QueryAnalysisPass(replacement_table_expression).run(filter, context);
     }
 
+    auto replacement_table_expression_ptr = replacement_table_expression;
     auto query_tree_to_modify = query_info.query_tree->cloneAndReplace(query_info.table_expression, std::move(replacement_table_expression));
```

Comment:
`IQueryTreeNode::cloneAndReplace` takes a `std::shared_ptr<IQueryTreeNode>` by value. I understand you might want to call `std::move` to avoid increments/decrements in the reference counter, but in your use case you are doing it in the previous line. Why don't you simply pass it by value and don't create an additional variable at all?